app/sagas/products.js

```diff
@@ -1,22 +1,48 @@
 import { put, call, takeLatest } from 'redux-saga/effects';
+import { Actions } from 'react-native-router-flux';
 
 import {
   requestProducts,
   productsSuccess,
   productsError,
+
+  requestCreateProduct,
+  createProductSuccess,
+  createProductError,
 } from '../reducers/products';
-import { fetchAllProducts } from '../Api';
+import * as Api from '../Api';
+import * as FirebaseApi from '../FirebaseApi';
 
 export function* fetchProducts() {
   try {
-    const response = yield call(fetchAllProducts);
-    const [, ...products] = response.data;
+    const { data } = yield call(Api.fetchAllProducts);
+    const products = Object.keys(data)
+      .map(id => ({ ...data[id], id }))
+      .reverse(); // new records first
```

Comment:
We definitely need to extract duplication from reducers and sagas, please create an issue for this.